packages/core/src/DirectoryLoader.ts

```diff
@@ -139,121 +149,124 @@ export abstract class DirectoryLoader {
 			this.types.nodes.push(description);
 		});
 
-		for (const credential of getCredentialsForNode(tempNode)) {
+		for (const credential of this.getCredentialsForNode(tempNode)) {
 			if (!this.nodesByCredential[credential.name]) {
 				this.nodesByCredential[credential.name] = [];
 			}
-			this.nodesByCredential[credential.name].push(fullNodeName);
+			this.nodesByCredential[credential.name].push(nodeType);
 		}
 	}
 
-	protected loadCredentialFromFile(credentialClassName: string, filePath: string): void {
-		let tempCredential: ICredentialType;
-		try {
-			tempCredential = loadClassInIsolation(filePath, credentialClassName);
+	protected loadCredentialFromFile(filePath: string): void {
+		const tempCredential = this.loadClass<ICredentialType>(filePath);
+		// Add serializer method "toJSON" to the class so that authenticate method (if defined)
+		// gets mapped to the authenticate attribute before it is sent to the client.
+		// The authenticate property is used by the client to decide whether or not to
+		// include the credential type in the predefined credentials (HTTP node)
+		Object.assign(tempCredential, { toJSON });
 
-			// Add serializer method "toJSON" to the class so that authenticate method (if defined)
-			// gets mapped to the authenticate attribute before it is sent to the client.
-			// The authenticate property is used by the client to decide whether or not to
-			// include the credential type in the predefined credentials (HTTP node)
-			Object.assign(tempCredential, { toJSON });
+		this.fixIconPaths(tempCredential, filePath);
 
-			this.fixIconPaths(tempCredential, filePath);
-		} catch (e) {
-			if (e instanceof TypeError) {
-				throw new ApplicationError(
-					'Class could not be found. Please check if the class is named correctly.',
-					{ extra: { credentialClassName } },
-				);
-			} else {
-				throw e;
-			}
-		}
-
-		this.known.credentials[tempCredential.name] = {
-			className: credentialClassName,
+		const credentialType = tempCredential.name;
+		this.known.credentials[credentialType] = {
+			className: tempCredential.constructor.name,
 			sourcePath: filePath,
 			extends: tempCredential.extends,
-			supportedNodes: this.nodesByCredential[tempCredential.name],
+			supportedNodes: this.nodesByCredential[credentialType],
 		};
 
-		this.credentialTypes[tempCredential.name] = {
+		this.credentialTypes[credentialType] = {
 			type: tempCredential,
 			sourcePath: filePath,
 		};
 
 		this.types.credentials.push(tempCredential);
 	}
 
+	private getCredentialsForNode(
+		object: IVersionedNodeType | INodeType,
+	): INodeCredentialDescription[] {
+		if ('nodeVersions' in object) {
+			return uniqBy(
+				Object.values(object.nodeVersions).flatMap(
+					(version) => version.description.credentials ?? [],
+				),
+				'name',
+			);
+		}
+
+		return object.description.credentials ?? [];
+	}
+
+	getVersionedNodeTypeAll(object: IVersionedNodeType | INodeType): INodeType[] {
```

Comment:
This is not very clear what this does. Could we add documentation?

Reply:
I also don't know. These were moved from [here](https://github.com/n8n-io/n8n/blob/master/packages/workflow/src/NodeHelpers.ts#L1969-L1986).
This code (and this PR in general) could definitely benefit from better docs, and more tests.